File: src/species-two.ts
import { Species } from './species.js';
import { Note } from './note.js';

export class SpeciesTwo extends Species {

	generateCounterpoint(cantusFirmus: Note[]): Note[] {
		return [];
	}
	
	chooseNextNote(): number {
		if ((this.noteTwoBefore - this.noteBefore) === 2) {
			return this.noteBefore + 1;
		} else if ((this.noteTwoBefore - this.noteBefore) === -2) {
			return this.noteTwoBefore + 1;
		} else if ((this.noteTwoBefore - this.noteBefore) === 1) {
			return this.noteTwoBefore + 1;
		} else if ((this.noteTwoBefore - this.noteBefore) === -1) {
			return this.noteTwoBefore - 1;
		} else if ((this.noteTwoBefore - this.noteBefore) < 0) {
			return this.noteBefore + 1;
		} else if ((this.noteTwoBefore - this.noteBefore) > 0) {
			return this.noteBefore - 1;
		} else {
			return this.noteTwoBefore + 1;
		}
	}
}

File: Music Project/SpeciesTwo.cpp
#include "SpeciesTwo.h"



SpeciesTwo::SpeciesTwo()
{
}


SpeciesTwo::~SpeciesTwo()
{
}

int SpeciesTwo::chooseNextNote() {
	if ((noteTwoBefore - noteBefore) == 2) {
		return noteBefore + 1;
	}
	else if ((noteTwoBefore - noteBefore) == -2) {
		return noteTwoBefore + 1;
	}
	else if ((noteTwoBefore - noteBefore) == 1) {
		// This would normally be an error but I'm making up rules because to fix the error would require rewriting from here on out of both the top and bottom line
		return noteTwoBefore + 1;
	}
	else if ((noteTwoBefore - noteBefore) == -1) {
		return noteTwoBefore - 1;
	}
	else if ((noteTwoBefore - noteBefore) < 0) {
		return noteBefore + 1;
	}
	else if ((noteTwoBefore - noteBefore) > 0) {
		return noteBefore - 1;
	}
	else {
		return noteTwoBefore + 1;
	}
}
